sphinx/ext/autodoc/__init__.py

```diff
@@ -673,22 +685,26 @@ def keyfunc(entry):
         for documenter, isattr in memberdocumenters:
             documenter.generate(
                 all_members=True, real_modname=self.real_modname,
-                check_module=members_check_module and not isattr)
+                check_module=members_check_module and not isattr,
+                additional_options=['abstract'])
```

Comment:
Why giving `abstract` option always?

Reply:
This was wrong or at least misleading. `additional_options` should have better been named `revealed_options`, and this only instructs the `Documenter` to include the `abstract` flag in the output *if it has been set in `self.options`*.

When you look at the current diff, this is being done in the `_revealed_header_options` property of `Documenter`, which is overridden in the `ClassLevelDocumenter`.